src/java/org/apache/nutch/crawl/Generator.java

```diff
@@ -838,12 +846,21 @@ public Generator(Configuration conf) {
       job.setOutputValueClass(CrawlDatum.class);
       FileOutputFormat.setOutputPath(job, tempDir2);
       try {
-        int complete = job.waitForCompletion(true)?0:1;
+        boolean success = job.waitForCompletion(true);
+        if (!success) {
+          String message = "Generator job did not succeed, job status:"
+              + job.getStatus().getState() + ", reason: "
+              + job.getStatus().getFailureInfo();
+          LOG.error(message);
+          NutchJob.cleanupAfterFailure(tempDir, lock, fs);
+          NutchJob.cleanupAfterFailure(tempDir2, lock, fs);
```

Comment:
ok, it's not an error to try to remove the lock file a second time.

Reply:
No @sebastian-nagel. AFAIK if I try to remove a lock file that does not exist, the function would just return False.